x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -585,7 +588,9 @@ class AgentPolicyService {
   public async delete(
     soClient: SavedObjectsClientContract,
     esClient: ElasticsearchClient,
-    id: string
+    id: string,
+    context: RequestHandlerContext,
+    request: KibanaRequest<unknown, unknown, unknown, any>
```

Comment:
I am not sure about adding this two parameter at the service level? why it's needed, it will make consuming the service outside of a request a lot harder

Reply:
Yes, you are right but they are needed by the externalCallback. The external callback needs to be executed after the `packagePolicyService.delete` action as it needs the response. 
We need this to execute this callback when the integration is removed in the policy remove action.
We are doing something similar in the package policy handler when it's removed.
Maybe @paul-tavares has other thoughts on this?

Reply:
I don't think the delete package callback needs the request or context, correct? if so, one option is to refactor the `PackagePolicyService` and split the running of those external callbacks to a different method (ex. `runExternalDeleteCallbacks()` - then you don't need to have these two dependencies to be required here.

The other option would be to create a specific extension point for deletion of Agent Policies (which I personally like better) and have that callback receive the agent policy deleted along with all of the package policies that were also deleted. This method here would remain untouched (maybe we need to improve the return value) and there would be another functions in this service to run the external callback.  That method (ex. `runDeleteExternalCallbacks()`) would then be called from the API route (and would also be available to the consumers of the AgentPolicyService). It would be the responsibility of the external callback to loop through the data to figure out if they should do something with it.

Reply:
The first option you suggest is probably doable without two much refactoring no? and having it here and not in the handler we ensure that the callback is well called, (if we delete a package policy from anywhere else in the future)

Reply:
yeah, it's not that much refactor (🤣 says the guys (me) that is not doing it LOL)

Reply:
You're right! context and request are not needed for the delete callback! 👍